frontend/src/components/NavBar.js

```diff
@@ -26,6 +34,26 @@ const NavLinks = styled.ul`
   display: flex;
   list-style: none;
   align-items: center;
+  position: relative;
+
+  @media (max-width: 950px) {
+    display: ${({ show }) =>
+      show ? "flex" : "none"}; /* Show/Hide based on state */
+    flex-direction: column;
+    align-items: center;
+    gap: 1rem;
+    position: absolute;
+    top: 0;
+    right: 0;
+    background-color: white;
+    width: auto;
+    height: 100vh;
+    transition: transform 0.3s ease-in-out;
+    transform: ${({ show }) => (show ? "translateX(0)" : "translateX(100%)")};
```

Comment:
Transition on the mobile menu (`translateX`) is a great addition! But, it may be beneficial to increase the duration slightly for a smoother effect.